### PR TITLE
Add fs.gs.create.items.conflict.check.enable property

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java [hunk 1]
```diff
@@ -231,8 +231,9 @@ public WritableByteChannel create(URI path, CreateFileOptions createOptions) thr
               "Cannot create a file whose name looks like a directory: '%s'", resourceId));
     }
 
-    // Before creating a leaf directory we need to check if there are no conflicting files
-    // with the same name as any subdirectory
+    // Before creating a file we need to check:
+    // - if there are no conflicting files with the same name as any parent subdirectory
+    // - if there are no conflicting directory with the name as a file
```
Comment (hongyegong) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java:236]: These comments are a bit confusing, what is "conflicting directory with the name as a file", is it file or directory? Should it be - check if there are no conflicting directories with the same name as the file being created?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java [hunk 2]
```diff
@@ -115,7 +116,9 @@
    * authority corresponding to that root path in the process.
    */
   public static final HadoopConfigurationProperty<Boolean> GCE_BUCKET_DELETE_ENABLE =
-      new HadoopConfigurationProperty<>("fs.gs.bucket.delete.enable", false);
+      new HadoopConfigurationProperty<>(
```
Comment (hongyegong) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java:119]: Do we really wanna do this? This looks to me less obvious by using a default. I feel we better unify the way to define these default values, not mixed.

**Review by hongyegong — CHANGES_REQUESTED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java (hunk 1 above)
Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java:237]: Clarified and added example.

**Review by medb — COMMENTED** [review]
Addressed comments.

**Review by hongyegong — APPROVED** [review]

**Pull request merged**